feat(SchemaField): render yaml ace editor

ahoskins opened the pull request:
This renders an `AceEditor` when "application/x-yaml" is encountered in the JSON schema media type. We assume the YAML is base64 encoded.

**Testing**:
1. Checkout this branch
2. Turn on fixtures
3. Navigate to catalog, and click "Review and Run" on any package, then click "Edit Config" and scroll down to the YAML in the form.

**After doing the above three steps you will see this:**
![screen shot 2017-11-22 at 2 06 19 pm](https://user-images.githubusercontent.com/1527504/33151777-5aebb508-cf8e-11e7-90bf-7f681113943a.png)

**If an invalid base64 encoded string is pasted into the JSON editor:**
![screen shot 2017-11-22 at 2 07 12 pm](https://user-images.githubusercontent.com/1527504/33151821-842f2de6-cf8e-11e7-9115-104ead817299.png)

**But since we don't do YAML validation, there are many base64 encoded strings that are not valid YAML, but will still render:**
![screen shot 2017-11-22 at 2 08 02 pm](https://user-images.githubusercontent.com/1527504/33151839-94fbef88-cf8e-11e7-822d-f0a065d93c13.png)

**If the field is required, it will show the required field error just like other form inputs:**
![screen shot 2017-11-22 at 2 00 43 pm](https://user-images.githubusercontent.com/1527504/33151679-e4464ed6-cf8d-11e7-9c96-9417b2425ade.png)

 

 



**Checklist**
- [ ] Did you add a JIRA issue in a commit message or as part of the branch name?
- [ ] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

<!-- More info can be found by clicking the "guidelines for contributing" link above. -->

Poltergeist commented on src/styles/components/framework-configuration/styles.less:
```diff
@@ -7,3 +7,14 @@
 }
+
+.framework-form-yaml-editor {
+  // todo use same border color as the inputs
```
`@form-control-border-color` might be what you are looking for.

ahoskins replied:
Nice, thanks 👍

ahoskins commented:
alfreddo test this

ahoskins commented:
After demo with the data services team and talking with @leemunroe, I've made some changes to this. 

1. Made it 100% width of it's container. This is more appropriate for editing bigger YAML files.
2. Added gutter which includes line numbers and folding.
3. 2 tab instead of 4 indent.
4. Font size normalized compared to the other font on the form. Other fonts are 15px, and this monospace font is 14px which makes it look same size as the other fonts.
5. Removed box shadow on scrolling, and the 80 character margin ruler thing.
